src/main/java/uk/gov/justice/hmpps/casenotes/services/CaseNoteService.java

```diff
@@ -40,36 +44,85 @@ public CaseNoteService(OffenderCaseNoteRepository repository, CaseNoteTypeReposi
         this.externalApiService = externalApiService;
     }
 
-    public Page<CaseNote> getCaseNotesByOffenderIdentifier(@NotNull final String offenderIdentifier, final Pageable pageable) {
-        final var filter = OffenderCaseNoteFilter.builder()
-                .offenderIdentifier(offenderIdentifier)
-                .build();
+    public Page<CaseNote> getCaseNotes(final String offenderIdentifier, final CaseNoteFilter caseNoteFilter, final Pageable pageable) {
 
-        return getCaseNotes(pageable, filter);
-    }
+        final var sensitiveCaseNotes = new ArrayList<CaseNote>();
 
-    public Page<CaseNote> getCaseNotes(final CaseNoteFilter caseNoteFilter, final Pageable pageable) {
-        final var filter = OffenderCaseNoteFilter.builder()
-                .type(caseNoteFilter.getType())
-                .subType(caseNoteFilter.getSubType())
-                .locationId(caseNoteFilter.getLocationId())
-                .staffUsername(caseNoteFilter.getStaffUsername())
-                .startDate(caseNoteFilter.getStartDate())
-                .endDate(caseNoteFilter.getEndDate())
-                .build();
+        if (SecurityUserContext.hasRoles("VIEW_SENSITIVE_CASE_NOTES", "ADD_SENSITIVE_CASE_NOTES")) {
```

Comment:
Am I right in thinking that if a user has the `ADD_SENSITIVE_CASE_NOTES` or `VIEW_SENSITIVE_CASE_NOTES` then they don't need a relationship with the offender (i.e. intersection of case loads)

Reply:
Good question!